### PR TITLE
chore(stylelint-badge): create property for forced-colors value to satisfy lint requirements

### PR DESCRIPTION
**Pull request by cdransf**
## Description

Assigns `CanvasText` to custom property to satisfy linting requirements.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/badge/index.css [hunk 1]
```diff
@@ -146,7 +148,7 @@
 /* windows high contrast mode */
 @media (forced-colors: active) {
 	.spectrum-Badge {
-		border-color: CanvasText; /* stylelint-disable-line declaration-property-value-no-unknown */
+		border-color:  var(--highcontrast-spectrum-badge-border-color);
```
Comment (5t3ph) [inline components/badge/index.css:151]: Giving a quick look at existing `--highcontrast-` custom properties, it looks like "spectrum" is usually dropped, like:

```suggestion
		border-color:  var(--highcontrast-badge-border-color);
```

**Review by 5t3ph — CHANGES_REQUESTED** [review]

File: components/badge/index.css (hunk 1 above)
Comment (cdransf) [reply components/badge/index.css]: Nice catch! Just updated that! :)

File: components/badge/index.css [hunk 2]
```diff
@@ -68,6 +68,8 @@
 	--spectrum-badge-icon-spacing-horizontal: var(--spectrum-component-edge-to-visual-100);
 	--spectrum-badge-icon-spacing-vertical-top: var(--spectrum-component-top-to-workflow-icon-100);
 	--spectrum-badge-icon-only-spacing-horizontal: var(--spectrum-component-edge-to-visual-only-100);
+
+	--highcontrast-spectrum-badge-border-color: CanvasText;
```
Comment (5t3ph) [inline components/badge/index.css:72]: Update needed here as well :)

```suggestion
	--highcontrast-badge-border-color: CanvasText;
```

File: components/badge/metadata/metadata.json [hunk 3]
```diff
@@ -177,5 +177,5 @@
   ],
   "system-theme": [],
   "passthroughs": [],
-  "high-contrast": []
+  "high-contrast": ["--highcontrast-spectrum-badge-border-color"]
```
Comment (5t3ph) [inline components/badge/metadata/metadata.json:180]: I'm not sure if it makes a difference in this case to edit the file directly, or if you need to re-run locally to have this updated to remove `spectrum`?

Comment (cdransf) [reply components/badge/metadata/metadata.json]: Good catch! In this case it was the only change, but I re-ran `yarn-ci` to verify. ✨

**Review by 5t3ph — APPROVED** [review]

**Pull request merged**